executor/join.go

```diff
@@ -529,17 +564,46 @@ func (e *NestedLoopApplyExec) prepare(goCtx goctx.Context) error {
 	}
 }
 
-func (e *NestedLoopApplyExec) doJoin(bigRow Row, match bool) ([]Row, error) {
+// fetchAllInners reads all data from the inner table and stores them in a slice.
+func (e *NestedLoopApplyExec) fetchAllInners(goCtx goctx.Context) error {
+	err := e.innerExec.Open(goCtx)
+	defer terror.Call(e.innerExec.Close)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	e.innerChunkRows = e.innerChunkRows[:0]
+	for {
+		chk := e.innerExec.newChunk()
```

Comment:
these allocated inner chunks can be reused in the next `fetchAllInners()` function call.

Reply:
So I created this PR https://github.com/pingcap/tidb/pull/5476
to make List can be used in joinResultGenerator, and a List can be easily reused.